fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/PartialDataV2RequestHandler.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web.handlers;
+
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+import com.yahoo.bard.webservice.druid.model.query.DruidAggregationQuery;
+import com.yahoo.bard.webservice.web.DataApiRequest;
+import com.yahoo.bard.webservice.web.responseprocessors.ResponseProcessor;
+
+import javax.validation.constraints.NotNull;
+
+/**
+ * A request handler that builds responses which filter partial data V2.
+ */
+public class PartialDataV2RequestHandler implements DataRequestHandler {
+
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+
+    private final DataRequestHandler next;
+
+    /**
+     * Constructor.
+     *
+     * @param next  Next Handler in the chain
+     */
+    public PartialDataV2RequestHandler(@NotNull DataRequestHandler next) {
+        this.next = next;
+    }
+
+    @Override
+    public boolean handleRequest(
+            RequestContext context,
+            DataApiRequest request,
+            DruidAggregationQuery<?> druidQuery,
+            ResponseProcessor response
+    ) {
+        return next.handleRequest(
+                context,
+                request,
+                druidQuery.withContext(
+                        druidQuery.getContext().withUncoveredIntervalsLimit(
+                                SYSTEM_CONFIG.getIntProperty(
+                                        SYSTEM_CONFIG.getPackageVariableName("druid_uncovered_interval_limit"),
+                                        0
+                                )
```

Comment:
This call should get moved up to construction time. We want to do as little work per-request as we can.